### PR TITLE
[exporter/datadog] Add `metrics::sums::cumulative_monotonic_mode` setting; deprecate `metrics::send_monotonic_counter`

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

- Add scaffolding for renaming configuration options.
- Add new `metrics::sums::cumulative_monotonic_mode` setting.
- Deprecate `metrics::send_monotonic_counter` in favor of `metrics::sums::cumulative_monotonic_mode`

**Link to tracking Issue:** #8489

**Testing:** 

- [x] Add unit tests to check that warning is emitted when using the deprecated setting
- [x] Add unit test to check that an error results from using deprecated + new at the same time

**Documentation:** Update the template

### CODE REVIEW

File: exporter/datadogexporter/config/warning_deprecated.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+// UpdateCfg to move the old configuration value into the new one.
+func (e renameError) UpdateCfg(cfg *Config) {
+	e.updateFn(cfg)
+}
+
+// handleRenamedSettings for a given configuration map.
+// Error out if any pair of old-new options are set at the same time.
+func handleRenamedSettings(configMap *config.Map, cfg *Config) (warnings []error, err error) {
+	for _, renaming := range renamedSettings {
+		ok, errCheck := renaming.Check(configMap)
```
Comment (KSerrania) [inline exporter/datadogexporter/config/warning_deprecated.go:87]: It's a bit weird for a boolean to be named `ok` if we log a warning when it's true 😄, but I'm not sure what would be a better fit here.

**Review by KSerrania — APPROVED** [review]
The deprecation system looks good to me.

File: exporter/datadogexporter/config/warning_deprecated.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+			} else {
+				c.Metrics.SumConfig.CumulativeMonotonicMode = CumulativeMonotonicSumModeRawValue
+			}
+		},
+	},
+}
+
+var _ error = (*renameError)(nil)
+
+// renameError is an error related to a renamed setting.
```
[the remark this answers is not in the document]
Comment (mx-psi) [reply exporter/datadogexporter/config/warning_deprecated.go:26]: Done

File: exporter/datadogexporter/config/warning_deprecated.go (hunk 1 above)
Comment (mx-psi) [reply exporter/datadogexporter/config/warning_deprecated.go]: I went with `isOldNameUsed`, it was weird indeed 😅

**Review by anuraaga — APPROVED** [review]

**Pull request merged**